src/wattsi.pas

```diff
@@ -504,9 +505,7 @@    TCrossReferences = record
 
                   Scratch := Default(Rope);
                   Scratch.Append('#');
-                  ExtractedData := Element.GetAttribute('id');
-                  Assert(not ExtractedData.IsEmpty);
```

Comment:
Add asserts anywhere you have an invariant and you're not confident the rest of the code will keep it, for example when someone breaks the code elsewhere. :-)